web/src/components/organisms/Asset/AssetList/index.tsx

```diff
@@ -30,53 +31,67 @@ const AssetList: React.FC = () => {
   };
   const hideUploadModal = () => {
     setUploadModalVisibility(false);
+    setUploading(false);
+    setFileList([]);
   };
 
   const handleUpload = () => {
     setUploading(true);
-    createAssets(fileList)
-      .catch(error => {
-        // TODO: notification
-        console.log(error);
-      })
-      .finally(() => {
-        setUploading(false);
-        setFileList([]);
-        hideUploadModal();
-      });
+    createAssets(fileList).finally(() => {
+      hideUploadModal();
+    });
   };
 
   const handleEdit = (asset: Asset) => {
     navigate(`/workspace/${workspaceId}/project/${projectId}/asset/${asset.id}`);
   };
 
+  const uploadProps: UploadProps = {
+    name: "file",
+    multiple: true,
+    directory: false,
+    showUploadList: true,
+    accept: imageFormats + "," + fileFormats,
+    listType: "picture",
+    onRemove: file => {
+      const index = fileList.indexOf(file);
+      const newFileList = fileList.slice();
+      newFileList.splice(index, 1);
+      setFileList(newFileList);
+    },
+    beforeUpload: (_file, files) => {
+      setFileList([...fileList, ...files]);
+      return false;
+    },
+    fileList,
+  };
+
   return (
-    <Wrapper>
-      <AssetListHeader
-        title="Asset"
-        fileList={fileList}
-        uploading={uploading}
-        uploadModalVisibility={uploadModalVisibility}
-        displayUploadModal={displayUploadModal}
-        hideUploadModal={hideUploadModal}
-        handleUpload={handleUpload}
-        setFileList={setFileList}
-      />
-      <AssetListTable
-        assetList={assetList}
-        assetsPerPage={assetsPerPage}
-        handleEdit={handleEdit}
-        handleSearchTerm={handleSearchTerm}
-        selection={selection}
-        setSelection={setSelection}
-      />
-    </Wrapper>
+    <AssetListBody
+      pageHeader={{
+        title: "Asset",
+        extra: (
+          <UploadAsset
+            fileList={fileList}
+            uploading={uploading}
+            uploadProps={uploadProps}
+            uploadModalVisibility={uploadModalVisibility}
+            displayUploadModal={displayUploadModal}
+            hideUploadModal={hideUploadModal}
+            handleUpload={handleUpload}
+          />
+        ),
+      }}
+      tableProps={{
+        assetList,
+        assetsPerPage,
+        handleEdit,
+        handleSearchTerm,
+        selection,
+        setSelection,
+      }}
```

Comment:
I think its better if you just pass the callbacks/variables declared in the organism in to the molecule. Then, in the molecule declare the table props and upload props.

Reply:
Done!